src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5278,15 +5278,16 @@ int RGWRados::restore_obj_from_cloud(RGWLCCloudTierCtx& tier_ctx,
   }
   boost::optional<RGWPutObj_Compress> compressor;
   CompressorRef plugin;
+  dest_placement.storage_class = tier_ctx.restore_storage_class;
   RGWRadosPutObj cb(dpp, cct, plugin, compressor, &processor, progress_cb, progress_data,
                     [&](map<string, bufferlist> obj_attrs) {
-                      // XXX: do we need filter() like in fetch_remote_obj() cb
+                      // XXX: do we need filter() lke in fetch_remote_obj() cb
                       dest_placement.inherit_from(dest_bucket_info.placement_rule);
                       /* For now we always restore to STANDARD storage-class.
                        * Later we will add support to take restore-target-storage-class
                        * for permanent restore
                        */
-                      dest_placement.storage_class = RGW_STORAGE_CLASS_STANDARD;
+  //                    dest_placement.storage_class = RGW_STORAGE_CLASS_STANDARD;
```

Comment:
what will be the storage in this case?

Reply:
It is now updated in Line:5281 above. A new option "restore_storage_class" is added (default:STANDARD)  to configure restored object storage-class.